app/models/blog.rb

```diff
@@ -6,16 +6,34 @@ class Blog < ApplicationRecord
   has_many :liking_users, class_name: 'User', source: :user, through: :likings
 
   validates :title, :content, presence: true
+  validate :eyecatch_requires_premium
 
   scope :published, -> { where('secret = FALSE') }
 
+  scope :visible_to, lambda { |user|
+    if user
+      where(secret: false).or(where(user_id: user.id))
+    else
+      where(secret: false)
+    end
```

Comment:
```suggestion
    published.or(where(user: user))
```

これでも同じ結果が得られそうです。（生成されるSQLを確認してみましょう）

あと細かいけど、こうすると()がネストしなくて少しだけ読みやすいかも？

```suggestion
    where(user: user).or(published)
```